### DISCUSSION TITLE
Does gha bill for data transfer?

### DISCUSSION THREAD

**Discussion by lackhoa** [community/community#27055] [Actions]
<p>For example, do I get billed for uploading / downloading artifact?</p>
<p>I saw this <a href="https://docs.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions" rel="noopener nofollow ugc">here</a>:</p>
<blockquote>
<p>At the end of the month, GitHub rounds your data transfer to the nearest GB.</p>
</blockquote>
<p>Don’t know what that’s for.</p>

**Comment by Simran-B** [comment]
<p>Is your repository public, or is it private but you use a self-hosted runner? Then no, it’s free. There are limitations for private repositories that use GitHub-hosted runners on a free plan, however.</p>
<blockquote>
<p>GitHub Actions usage is free for both public repositories and self-hosted runners. For private repositories, each GitHub account receives a certain amount of free minutes and storage […]</p>
</blockquote>
<p>In the latter case, you get 500 MB for free. If you need more, then you have to pay. Artifacts do count as storage:</p>
<blockquote>
<p>The storage used by a repository is the total storage used by GitHub Actions artifacts and GitHub Packages.</p>
</blockquote>

**Comment by lackhoa** [comment]
<p>I just want to clarify the question: it’s about data transfer, not storage. I get the impression that you’re talking about storage (In fact you do mention storage, but if that relates to data transfer somehow then please tell me).</p>

**Comment by Simran-B** [comment]
<p>My bad. It’s peculiar that the docs mention data transfer once on the page, but nothing else about traffic. Looking at <a href="https://docs.github.com/en/billing/managing-billing-for-github-actions/viewing-your-github-actions-usage" class="inline-onebox" rel="noopener nofollow ugc">Viewing your GitHub Actions usage - GitHub Docs</a> I get the impression “data transfer” refers to the minutes spent?!</p>
<p>I vaguely remember some footnote that said that internal traffic (which would include artifacts) is included for free, but I can’t find it anymore and also nothing else about network egress. You may want to reach out to sales to get clarity on this.</p>

**Comment by lackhoa** [comment]
<p>Thanks for clarifying, I’ll ask sales and post the reply here when I have time.</p>

**Comment by lackhoa** [comment, marked as answer]
<p>This is the reply from support</p>
<blockquote>
<p>All data transferred out, when triggered by GitHub Actions, and data transferred in from any source is free. We determine you are downloading packages using GitHub Actions when you log in to GitHub Packages using a <code>GITHUB_TOKEN</code> .<br>
Your account on the free plan receives 1 GB of free included data transfer per month.<br>
<a href="https://docs.github.com/en/billing/managing-billing-for-github-packages/about-billing-for-github-packages" class="inline-onebox" rel="noopener nofollow ugc">About billing for GitHub Packages - GitHub Docs</a></p>
</blockquote>
<p>So there you go: it’s not billed.</p>

**Comment by Simran-B** [comment]
<p>Thanks! As per the docs, GitHub Packages is also free for public repos and you get 1 GB / month of egress traffic for private repos, but GitHub Actions downloading your private GitHub Packages does not count towards that limit (as long as you use the <code>GITHUB_TOKEN</code>).</p>
<p>I opened a docs issue: <a href="https://github.com/github/docs/issues/7569" class="inline-onebox" rel="noopener nofollow ugc">GitHub Actions data transfer is not billed, but mentioned in the docs · Issue #7569 · github/docs · GitHub</a></p>

**Comment by Simran-B** [comment]
<p>Found the statement that I remembered on this page: <a href="https://github.com/features/packages#pricing" class="inline-onebox" rel="noopener nofollow ugc">GitHub Packages: Your packages, at home with their code · GitHub</a></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/5/2/52de2d23fe9637d3d6e2ab24614048f82ddfd7f9.png" data-download-href="/uploads/short-url/bP5a7dRZBVA3ayjFsnPsRtoA4c1.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/7819991/181101512-e69f71ab-cdc8-4cf8-961f-0b516d2eb41b.png" alt="image" data-base62-sha1="bP5a7dRZBVA3ayjFsnPsRtoA4c1" width="690" height="141" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/5/2/52de2d23fe9637d3d6e2ab24614048f82ddfd7f9_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">835×171 9.42 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
